data/virt_autotest/guest_console_monitor.sh

```diff
@@ -0,0 +1,240 @@
+}
+
+## -1 means never timeout
+set timeout -1
+
+expect -re {dev.*\s([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})[^0-9]}
+
+if {![info exists expect_out(1,string)]} {
+        puts \"Match did not happen :(\"
+        exit 1
+}
+set output \$expect_out(buffer)
+expect eof
+
+puts \"\$output\"
+"
+
+}
+
+
+function open_console()
+{
+        #open guest console
+        for vmguest in ${vm_guestnames_state_running_array[@]};do
+                get_console ${vmguest} >> ${LOG_DIR}/${vmguest}_console_${timestamp} 2>&1 &
+        done 
+}
+
+function close_console()
+{
+        #close guest console
+        echo "kill PIDs $(basename $0)"
+        pid=`ps aux | grep "$(basename $0) -s" | grep -v "grep"| awk '{print $2}'`
+        if [[ -n $pid ]];then
+            kill -9 $pid >> /dev/null 2>&1;sync
+        else
+            echo "PIDs $(basename $0) were killed"
+        fi        
+        
+        echo "Kill all associated "virsh console" processes"
+        ps aux | grep "virsh console" | grep -v "grep" >> /dev/null 2>&1
+        while [[ $? -eq 0 ]];    
+        do
+            #wait for checking all associated "virsh console" processes"
+            sleep 3
+            sub_pid=`ps aux | grep "expect -c" | grep -v "grep"| awk '{print $2}'`
+            if [[ -n $sub_pid ]];then 
+                kill -9 $sub_pid >> /dev/null 2>&1;sync
+            else
+                break
+            fi    
+        done        
+}
+
+
+
+function end_monitor()
+{
+        #end of monitor for guest console
+	if [ -f ${MONITOR_LOCKED_FILE} ];then
+            echo "End of Monitor for guest console"	
+            #UNLOCK Monitor for guest console	    
+            > ${MONITOR_LOCKED_FILE} ; rm -rf ${MONITOR_LOCKED_FILE}
+            #Kill Pids from Monitor for guest console
+            close_console
+        fi
+}
+
+function start_monitor()
+{
+        #start monitor for guest console
+        get_vm_guestnames_state_running=`virsh list --state-running | grep "${vm_guestnames_types}" | awk '{print $2}'`
+        vm_guestnames_state_running_array=$(echo -e ${get_vm_guestnames_state_running})  
+
+        while :
+        do
+		if [ -f ${MONITOR_LOCKED_FILE} ];then
+                #Check with LOCK-UP status
+                cat ${MONITOR_LOCKED_FILE} | grep "LOCKED"                
+                if [ $? -eq 0 ];then
+                    #KEEP LOCK-UP status
+                    #To let just only one virsh console with one vm guest as backgroup PID			        
+                    
+                    #wait for checking background PID of virsh console
+                    sleep 3			       
+			        #Check with background PID of virsh console
+			        for vmguest in ${vm_guestnames_array[@]};do
+			            virsh list --state-running | grep ${vmguest}
+			                if [[ $? -eq 0 ]];then
+			                    ps aux | grep "virsh console ${vmguest}" | grep -v grep
+                                if [[ $? -ne 0 ]];then
```

Comment:
not aligned, and later lines in this function also have unaligned lines

Reply:
updated as aligned